api/agent/agent.go

```diff
@@ -487,6 +488,16 @@ func (a *agent) launch(ctx context.Context, slots chan<- slot, call *call, tok R
 	ch := make(chan error, 1)
 
 	if !protocol.IsStreamable(protocol.Protocol(call.Format)) {
+
+		deadline, err := common.GetRemainingTime(ctx)
+		if err != nil {
+			tok.Close()
+			ch <- err
+			return ch
+		}
+
+		call.EnvVars["FN_DEADLINE"] = strconv.FormatFloat(deadline.Seconds(), 'f', 0, 64)
```

Comment:
I think this is faulty because there's some latency between the user getting a handle on this and when we're setting it. what do you think about just sticking an RFC3339 time stamp in `FN_DEADLINE`, then there's no races, but I am uncertain how easy it is to use vs seconds across all languages (I suppose that users will do `time.Now().Add(fnDeadlineSeconds * time.Second)` to get the actual timeout in their code, in whatever lang they're using, am less sure of non-go language patterns for setting time based deadlines)

Reply:
hmm, goofed again, this is unix time. that works. i'm gonna get coffee, brain not working

Reply:
nvm, since we're converting to duration it's not unix time. okay, actually coffee now (sorry)

Reply:
RFC3339 sounds good, then I can just set it during instantiation just like 'CreatedAt'